test/path_filtering.go

```diff
@@ -0,0 +1,83 @@
+// +build examples
+
+/*
+Copyright 2021 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+type pathFilter func(string) bool
+
+// getPathFilter returns a pathFilter that filters out examples
+// unsuitable for the current feature-gate. For example,
+// if the enable-api-fields feature flag is currently set
+// to "alpha" then all "stable" and "beta" examples would be
+// excluded. Similarly when the flag is set to "stable",
+// all "alpha" and "beta" examples are filtered out.
```

Comment:
I am trying to understand this, please bear with me.

If I am reading this right, the feature flag value is considered as absolute and only allows examples of that particular version to be tested.

But based on the TEP, setting the feature gate to `alpha` means the controller must support `alpha` and `stable` both:

| Feature Versions ->  | beta | alpha |
| ---  | --- | --- |
| stable | x | |
| alpha | x | x |

x == "**enabled**"

Should we run stable examples with alpha feature gate?  Am I mixing things up here? 🙃  Is this meant to work with something else? 🤔

Reply:
I think you're absolutely right here.  I've updated the func to run both alpha and stable examples when enable-api-fields is "alpha".